### ISSUE TITLE
Scala Hadoop Shred: get the CustomPartitionSourceTest working with Hadoop 2.4

### ISSUE THREAD

**Issue by fblundun** [snowplow/snowplow#1960]
It is being removed from r71 to prevent `sbt assembly` from failing, but can be found in the project history.

This is the error:

```
[info] PartitionedTsv fed a ShreddedPartition should
[error] ! split output by the schema path
[error]     FlowException: unhandled exception (BaseFlow.java:918)
[error] cascading.flow.BaseFlow.complete(BaseFlow.java:918)
[error] com.twitter.scalding.Job.run(Job.scala:296)
[error] com.twitter.scalding.JobTest.runJob(JobTest.scala:201)
[error] com.twitter.scalding.JobTest.runHadoop(JobTest.scala:155)
[error] com.snowplowanalytics.snowplow.enrich.hadoop.outputs.CustomPartitionSourceTest$$anonfun$1$$anonfun$apply$1.apply(ShreddedPartitionSpec.scala:68)
[error] com.snowplowanalytics.snowplow.enrich.hadoop.outputs.CustomPartitionSourceTest$$anonfun$1$$anonfun$apply$1.apply(ShreddedPartitionSpec.scala:53)
[error] org.apache.hadoop.mapreduce.Cluster.initialize(Cluster.java:120)
[error] org.apache.hadoop.mapreduce.Cluster.<init>(Cluster.java:82)
[error] org.apache.hadoop.mapreduce.Cluster.<init>(Cluster.java:75)
[error] org.apache.hadoop.mapred.JobClient.init(JobClient.java:470)
[error] org.apache.hadoop.mapred.JobClient.<init>(JobClient.java:449)
[error] cascading.flow.hadoop.planner.HadoopFlowStepJob.internalNonBlockingStart(HadoopFlowStepJob.java:107)
[error] cascading.flow.planner.FlowStepJob.blockOnJob(FlowStepJob.java:236)
[error] cascading.flow.planner.FlowStepJob.start(FlowStepJob.java:162)
[error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:124)
[error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:43)
[error] Cannot initialize Cluster. Please check your configuration for mapreduce.framework.name and the correspond server addresses.
[error] org.apache.hadoop.mapreduce.Cluster.initialize(Cluster.java:120)
[error] org.apache.hadoop.mapreduce.Cluster.<init>(Cluster.java:82)
[error] org.apache.hadoop.mapreduce.Cluster.<init>(Cluster.java:75)
[error] org.apache.hadoop.mapred.JobClient.init(JobClient.java:470)
[error] org.apache.hadoop.mapred.JobClient.<init>(JobClient.java:449)
[error] cascading.flow.hadoop.planner.HadoopFlowStepJob.internalNonBlockingStart(HadoopFlowStepJob.java:107)
[error] cascading.flow.planner.FlowStepJob.blockOnJob(FlowStepJob.java:236)
[error] cascading.flow.planner.FlowStepJob.start(FlowStepJob.java:162)
[error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:124)
[error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:43)

```

**Comment by alexanderdean** [comment]
Can you link to the cascading-user group thread?

**Comment by fblundun** [comment]
Here it is: https://groups.google.com/forum/#!topic/cascading-user/GwlSn7DFYaU

**Comment by chuwy** [comment]
Test now fails only on Travis CI with following exception:

```
   - [error] ! split output by the schema path
   - [error]     FlowException: unhandled exception (BaseFlow.java:918)
   - [error] cascading.flow.BaseFlow.complete(BaseFlow.java:918)
   - [error] com.twitter.scalding.Job.run(Job.scala:296)
   - [error] com.twitter.scalding.JobTest.runJob(JobTest.scala:201)
   - [error] com.twitter.scalding.JobTest.runHadoop(JobTest.scala:155)
   - [error] com.snowplowanalytics.snowplow.enrich.hadoop.outputs.CustomPartitionSourceTest$$anonfun$1$$anonfun$apply$1.apply(CustomPartitionSourceTest.scala:69)
   - [error] com.snowplowanalytics.snowplow.enrich.hadoop.outputs.CustomPartitionSourceTest$$anonfun$1$$anonfun$apply$1.apply(CustomPartitionSourceTest.scala:53)
   - [error] org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:350)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1285)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1282)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapreduce.Job.submit(Job.java:1282)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:562)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:557)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapred.JobClient.submitJobInternal(JobClient.java:557)
   - [error] org.apache.hadoop.mapred.JobClient.submitJob(JobClient.java:548)
   - [error] cascading.flow.hadoop.planner.HadoopFlowStepJob.internalNonBlockingStart(HadoopFlowStepJob.java:108)
   - [error] cascading.flow.planner.FlowStepJob.blockOnJob(FlowStepJob.java:236)
   - [error] cascading.flow.planner.FlowStepJob.start(FlowStepJob.java:162)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:124)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:43)
   - [error] org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:350)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1285)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1282)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapreduce.Job.submit(Job.java:1282)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:562)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:557)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapred.JobClient.submitJobInternal(JobClient.java:557)
   - [error] org.apache.hadoop.mapred.JobClient.submitJob(JobClient.java:548)
   - [error] cascading.flow.hadoop.planner.HadoopFlowStepJob.internalNonBlockingStart(HadoopFlowStepJob.java:108)
   - [error] cascading.flow.planner.FlowStepJob.blockOnJob(FlowStepJob.java:236)
   - [error] cascading.flow.planner.FlowStepJob.start(FlowStepJob.java:162)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:124)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:43)
   - [error] short-hostname: short-hostname
   - [error] org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:350)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1285)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1282)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapreduce.Job.submit(Job.java:1282)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:562)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:557)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapred.JobClient.submitJobInternal(JobClient.java:557)
   - [error] org.apache.hadoop.mapred.JobClient.submitJob(JobClient.java:548)
   - [error] cascading.flow.hadoop.planner.HadoopFlowStepJob.internalNonBlockingStart(HadoopFlowStepJob.java:108)
   - [error] cascading.flow.planner.FlowStepJob.blockOnJob(FlowStepJob.java:236)
   - [error] cascading.flow.planner.FlowStepJob.start(FlowStepJob.java:162)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:124)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:43)
   - [error] short-hostname
   - [error] org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:350)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1285)
   - [error] org.apache.hadoop.mapreduce.Job$10.run(Job.java:1282)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapreduce.Job.submit(Job.java:1282)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:562)
   - [error] org.apache.hadoop.mapred.JobClient$1.run(JobClient.java:557)
   - [error] javax.security.auth.Subject.doAs(Subject.java:415)
   - [error] org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1556)
   - [error] org.apache.hadoop.mapred.JobClient.submitJobInternal(JobClient.java:557)
   - [error] org.apache.hadoop.mapred.JobClient.submitJob(JobClient.java:548)
   - [error] cascading.flow.hadoop.planner.HadoopFlowStepJob.internalNonBlockingStart(HadoopFlowStepJob.java:108)
   - [error] cascading.flow.planner.FlowStepJob.blockOnJob(FlowStepJob.java:236)
   - [error] cascading.flow.planner.FlowStepJob.start(FlowStepJob.java:162)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:124)
   - [error] cascading.flow.planner.FlowStepJob.call(FlowStepJob.java:43)
```

I'm bit suspicious about these lines:

```
short-hostname: short-hostname
```

This is host name on Travis CI/CD set to fix #2447, which is not relevant any more. I'll remove this line and if this is gonna work - I'll move https://github.com/snowplow/snowplow/issues/2674 to current milestone.
